jf_agent/git/gitlab_client.py

```diff
@@ -0,0 +1,99 @@
+import gitlab
+import logging
+import requests
+
+logger = logging.getLogger(__name__)
+
+
+class GitLabClient:
+    def __init__(self, server_url, private_token, per_page_override, session):
+        kwargs = {'private_token': private_token, 'session': session}
+        if per_page_override is not None:
+            kwargs['per_page'] = per_page_override
+        self.client = gitlab.Gitlab(server_url, **kwargs)
+
+    @staticmethod
+    def _get_diff_string(merge_request):
+        changes = merge_request.changes()
+        diffs = [change['diff'] for change in changes['changes']]
+        return '\n'.join(diffs)
+
+    def expand_merge_request_data(self, merge_request, target_project=None):
```

Comment:
Should expand_merge_request_data and _get_diff_string be in gitlab_adapter.py rather than gitlab_client.py?

Reply:
To me the adapter is in the context of the end result (data being massaged into its final form) whereas the client is a collection of convenience methods in the context of GitLab.  To me `_get_diff_string` and `expand_merge_request_data` falls into the latter for this reason. 

I think keeping these separate makes the context more clear as well, considering the adapter and the client have conflicting nomenclatures as well.